src/ConstantProduct.sol

```diff
@@ -63,6 +68,21 @@ contract ConstantProduct is IConditionalOrderGenerator {
      * @notice The second of the two tokens traded by this AMM.
      */
     IERC20 public immutable token1;
+    /**
+     * @notice The address that can execute administrative tasks on this AMM,
+     * as for example enabling/disabling trading or withdrawing funds.
+     */
+    address public immutable manager;
+    /**
+     * @notice The domain separator used for hashing CoW Protocol orders.
+     */
+    bytes32 public immutable solutionSettlerDomainSeparator;
+    /**
+     * @notice The address that needs to be approved for each token to trade
+     * on CoW Protocol.
+     */
+    address public immutable vaultRelayer;
+
```

Comment:
nit: not sure if `vaultRelayer` state is ultimately needed

Reply:
Good point, it's only needed for the first approval. Removed.